sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/state/AsynchronousMetricStorage.java

```diff
@@ -143,7 +143,17 @@ public MetricData collectAndReset(
             collectionInfo.getSupportedAggregation(), collectionInfo.getPreferredAggregation());
     collectLock.lock();
     try {
-      metricUpdater.run();
+      try {
+        metricUpdater.run();
+      } catch (RuntimeException e) {
+        logger.log(
+            Level.WARNING,
+            "An exception occurred invoking callback for instrument "
+                + getMetricDescriptor().getName()
+                + ".",
+            e);
+        return null;
```

Comment:
This should return `MetricData.empty()` once `@Nullable` annotation is removed in #3887.